plugins/redis/redis.cpp

```diff
@@ -1284,77 +1445,86 @@ void Connection::handleLockOnGet(ICodeContext * ctx, const char * key, MemoryAtt
-        struct timeval to = { 10, 0 };//10secs
-        ::redisSetTimeout(subscriptionConnection->context, to);
+            struct timeval to = { 10, 0 };//10secs
+            ::redisSetTimeout(subscriptionConnection->context, to);
 #endif
 
-        OwnedReply subReply = new Reply();
-        subscriptionConnection->readReply(subReply);
-        subscriptionConnection->assertOnErrorWithCmdMsg(subReply->query(), "GetOrLock<type>", key);
+            OwnedReply subReply = new Reply();
+            subscriptionConnection->readReply(subReply);
+            subscriptionConnection->assertOnErrorWithCmdMsg(subReply->query(), "GetOrLock<type>", key);
 
-        if (subscriptionConnection->isCorrectChannel(subReply->query(), "message"))
-        {
-            //We are about to return a value, to conform with other Get<type> functions, fail if the key did not exist.
-            //Since the value is sent via a published message, there is no direct reply struct so assume that an empty
-            //string is equivalent to a non-existent key.
-            //More importantly, it is paramount that this routine only return an empty string under one condition, that
-            //which indicates to the caller that the key was successfully locked.
-            //NOTE: it is possible for an empty message to have been PUBLISHed.
-            if (subReply->query()->element[2]->len > 0)
+            if (subscriptionConnection->isCorrectChannel(subReply->query(), "message"))
             {
-                retVal->set(subReply->query()->element[2]->len, subReply->query()->element[2]->str);//return the published value rather than another (WATCHed) GET.
-                return;
+                //We are about to return a value, to conform with other Get<type> functions, fail if the key did not exist.
+                //Since the value is sent via a published message, there is no direct reply struct so assume that an empty
+                //string is equivalent to a non-existent key.
+                //More importantly, it is paramount that this routine only return an empty string under one condition, that
+                //which indicates to the caller that the key was successfully locked.
+                //NOTE: it is possible for an empty message to have been PUBLISHed.
+                if (subReply->query()->element[2]->len > 0)
+                {
+                    retVal->set(subReply->query()->element[2]->len, subReply->query()->element[2]->str);//return the published value rather than another (WATCHed) GET.
+                    return;
+                }
+                //fail that key does not exist
+                redisReply fakeReply;
+                fakeReply.type = REDIS_REPLY_NIL;
+                assertKey(&fakeReply, key);
             }
-            //fail that key does not exist
-            redisReply fakeReply;
-            fakeReply.type = REDIS_REPLY_NIL;
-            assertKey(&fakeReply, key);
         }
+        throwUnexpected();
```

Comment:
This code appears to be reachable from the above... should give a better error message if so